std/js/_std/Std.hx

```diff
@@ -45,7 +45,7 @@ import js.Boot;
 	public static function parseInt( x : String ) : Null<Int> {
 		var v = untyped __js__("parseInt")(x, 10);
 		// parse again if hexadecimal
-		if( v == 0 && (x.charCodeAt(1) == 'x'.code || x.charCodeAt(1) == 'X'.code) )
+		if( v == 0 && untyped __js__('typeof {0}=="string"',x) && (x.charCodeAt(1) == 'x'.code || x.charCodeAt(1) == 'X'.code) )
```

Comment:
It's recommended to use [`js.Syntax.code()`](https://api.haxe.org/v/development/js/Syntax.html#code) instead of `untyped __js__()` now.

Reply:
Fascinating, this is the first I've heard of it. Happy to use either, though since the function already uses `untyped __js__` I'd be tempted to say updating to `js.Syntax.code` is a separate issue.